From: Hilko Bengen <bengen@hilluzination.de>
Date: Tue, 11 Aug 2015 22:04:25 +0200
Subject: Fix shell scripts

stenocurl: Parameter quoting, useless use of cat
stenoread: Do a better job finding tcpdump, stenocurl
---
 stenocurl | 8 ++++----
 stenoread | 7 +++++--
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/stenocurl b/stenocurl
index a5b3861..4ffab21 100755
--- a/stenocurl
+++ b/stenocurl
@@ -25,7 +25,7 @@ if [ -z "$JQ" ]; then
   exit 1
 fi
 
-if [ "$#" -lt 1 -o ${1:0:1} != "/" ]; then
+if [ "$#" -lt 1 -o "${1:0:1}" != "/" ]; then
   /bin/cat >&2 <<EOF
 USAGE: $0 /<path> [curl args...]
 
@@ -53,9 +53,9 @@ EOF
   exit 1
 fi
 
-HOST="$(/bin/cat "$STENOGRAPHER_CONFIG" | $JQ -r '.Host')"
-PORT="$(/bin/cat "$STENOGRAPHER_CONFIG" | $JQ -r '.Port')"
-CERTPATH="$(/bin/cat "$STENOGRAPHER_CONFIG" | $JQ -r '.CertPath')"
+HOST="$( < "$STENOGRAPHER_CONFIG" $JQ -r '.Host')"
+PORT="$( < "$STENOGRAPHER_CONFIG" $JQ -r '.Port')"
+CERTPATH="$( < "$STENOGRAPHER_CONFIG" $JQ -r '.CertPath')"
 if [ -z "$PORT" -o -z "$CERTPATH" ]; then
   echo "Unable to get port ($PORT) or certpath ($CERTPATH) from config ($STENOGRAPHER_CONFIG)" >&2
   exit 1
diff --git a/stenoread b/stenoread
index eb932e5..04296b3 100755
--- a/stenoread
+++ b/stenoread
@@ -38,10 +38,13 @@ fi
 STENOQUERY="$1"
 shift
 
+TCPDUMP=$(PATH=$PATH:/usr/local/sbin:/usr/sbin:/sbin which tcpdump)
+STENOCURL=$(PATH=$(dirname "$0"):$PATH which stenocurl)
+
 echo "Running stenographer query '$STENOQUERY', piping to 'tcpdump $@'" >&2
-$(dirname $0)/stenocurl /query \
+"$STENOCURL" /query \
     -d "$STENOQUERY" \
     --silent \
     --max-time 890 \
     --show-error |
-    tcpdump -r /dev/stdin -s 0 "$@"
+    "$TCPDUMP" -r /dev/stdin -s 0 "$@"
